src/Infrastructure/TimeMgr/tests/ESMF_AlarmUTest.F90

```diff
@@ -3598,9 +1606,1370 @@ subroutine Test_AlarmHang(testPass, rc)
+    if(status /= ESMF_SUCCESS) call ESMF_Finalize()
+    if (.not. ringing) then
+      allCorrect = .false.
+    end if
+    do n = 1, 6
+      call ESMF_ClockAdvance(clock,rc=status)
+      if(status /= ESMF_SUCCESS) call ESMF_Finalize()
+      call ESMF_AlarmGet(alarm, ringing=ringing, enabled=enabled, sticky=sticky, rc=status)
+      if(status /= ESMF_SUCCESS) call ESMF_Finalize()
+      if (modulo(n,2) == 0) then
+        ! Alarm should be ringing on even time steps
+        if (.not. ringing) then
+          allCorrect = .false.
+        end if
+      else
+        ! Alarm should NOT be ringing on odd time steps
+        if (ringing) then
+          allCorrect = .false.
+        end if
+      end if
+    enddo
+
+    if (allCorrect) then
+      rc = ESMF_SUCCESS
+    end if
+  end subroutine
+!------------------------------------------------------------------------
+  subroutine ForwardAlarm_Test3(rc)
+    ! Test 3: c.t0 = a.t0, c.dt = 20 min, a.dt = 10 min
+    !
+    ! This should raise an error: it's an error to create an alarm with alarm timeInterval
+    ! less than clock timeStep
+
+    integer, intent(out) :: rc
+
+    type (ESMF_Clock) :: clock
+    type (ESMF_Alarm) :: alarm
+    type (ESMF_Time)  :: startTime, stopTime, ringTime
+    type (ESMF_TimeInterval)  :: timeStep, ringTimeInterval
+
+    integer           :: status, n
+    logical           :: ringing, enabled, sticky
+
+    rc = ESMF_FAILURE
+
+    call ESMF_TimeSet(startTime,yy=1,mm=1,dd=1,h=0,m=0,s=0,rc=status)
+    if(status /= ESMF_SUCCESS) call ESMF_Finalize()
+    call ESMF_TimeSet(stopTime,yy=1,mm=1,dd=1,h=1,m=0,s=0,rc=status)
+    if(status /= ESMF_SUCCESS) call ESMF_Finalize()
+    call ESMF_TimeSet(ringTime,yy=1,mm=1,dd=1,h=0,m=0,s=0,rc=status)
+    if(status /= ESMF_SUCCESS) call ESMF_Finalize()
+
+    call ESMF_TimeIntervalSet(timeStep,S=1200, sN=0, sD=1,rc=status)
+    if(status /= ESMF_SUCCESS) call ESMF_Finalize()
+    call ESMF_TimeIntervalSet(ringTimeInterval,S=600, sN=0, sD=1,rc=status)
+    if(status /= ESMF_SUCCESS) call ESMF_Finalize()
+
+    clock = ESMF_ClockCreate(timeStep=timeStep,startTime=startTime,stopTime=stopTime,rc=status)
+    if(status /= ESMF_SUCCESS) call ESMF_Finalize()
+
+    ! It's an error to create an alarm with alarm timeInterval less than clock timeStep
+    alarm = ESMF_AlarmCreate(clock, ringTime=ringTime, ringInterval=ringTimeInterval, enabled=.true., sticky=.false., name='alarm', rc=status)
+    if(status /= ESMF_SUCCESS) then
+      ! This is the expected result: the status of ESMF_AlarmCreate should be an error, so
+      ! in this case, this test returns SUCCESS
+      rc = ESMF_SUCCESS
+    else
+      ! On the other hand, if ESMF_AlarmCreate succeeded, then this test has failed
+      rc = ESMF_FAILURE
+    end if
+    !call ESMF_AlarmDebug(alarm,'test3',rc=status)
+  end subroutine
+!------------------------------------------------------------------------
+  subroutine ForwardAlarm_Test4(rc)
```

Comment:
- [ ] The current behavior as demonstrated in this test seems problematic and unintuitive, especially for sticky alarms: As demonstrated here, when the alarm dt doesn't align with the clock dt, the alarm will only ring when the two come into alignment, even for a sticky alarm. This is (I think) in contrast to the previous behavior, when the alarm would ring when we first pass the ring time.